File: 0476-number-complement/0476-number-complement.py
class Solution:
    def findComplement(self, num: int) -> int:
        # Create a mask with all bits set to 1 for the length of num
        mask = 1
        while mask < num:
            mask = (mask << 1) | 1
        
        # Return the complement using XOR
        return num ^ mask

File: 0476-number-complement/0476-number-complement.js
/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function(num) {
    // Create a mask with all bits set to 1 for the length of num
    let mask = 1;
    while (mask < num) {
        mask = (mask << 1) | 1;
    }
    
    // Return the complement using XOR
    return num ^ mask;
};
